modular_citadel/code/modules/reagents/reagent container/hypospraymkii.dm

```diff
@@ -154,32 +167,32 @@
 					if(L != user)
 						L.visible_message("<span class='danger'>[user] is trying to inject [L] with [src]!</span>", \
 										"<span class='userdanger'>[user] is trying to inject [L] with the [src]!</span>")
-						if(!do_mob(user, L, extra_checks=CALLBACK(L, /mob/living/proc/can_inject,user,1)))
+						if(!do_mob(user, L, inject_wait, extra_checks=CALLBACK(L, /mob/living/proc/can_inject,user,1)))
 							return
-						if(!reagents.total_volume)
+						if(!vial.reagents.total_volume)
 							return
 						if(L.reagents.total_volume >= L.reagents.maximum_volume)
 							return
 						L.visible_message("<span class='danger'>[user] uses the [src] on [L]!</span>", \
 										"<span class='userdanger'>[user] uses the [src] on [L]!</span>")
 					else
-						if(!do_mob(user, L, extra_checks=CALLBACK(L, /mob/living/proc/can_inject,user,1)))
+						if(!do_mob(user, L, 0, extra_checks=CALLBACK(L, /mob/living/proc/can_inject,user,1)))
```

Comment:
Self injection needs to have a wait time, else it'll surpass a box of synthflesh patch meta that was happening before.

Reply:
Self spray, but I get your concern.